cfme/tests/containers/test_smart_management.py

```diff
@@ -1,123 +1,92 @@
+    if unique_items:
+        # Return a random item
+        selected_item_name = random.choice(unique_items)
+        return mapping[selected_item_name]
+    else:
+        pytest.skip("No unique item was found")
 
 
-@pytest.mark.parametrize('test_item', TEST_ITEMS,
-                         ids=[ContainersTestItem.get_pretty_id(ti) for ti in TEST_ITEMS])
 @pytest.mark.meta(blockers=[BZ(1479412,
                                forced_streams=['5.7'],
                                unblock=lambda test_item: test_item.obj != Container)])
-def test_smart_management_add_tag(provider, appliance, test_item):
-    logger.debug("Setting smart mgmt tag to {obj_type}".format(obj_type=test_item.obj.__name__))
+@pytest.fixture
+def get_clean_entity(**kwargs):
+    collection = kwargs.get("collection", None)
+    entity = kwargs.get("entity") if "entity" in kwargs else get_entity(collection)
+
+    tags = entity.get_tags()
+    entity.remove_tags(tags)
+    return entity
+
+
+@pytest.mark.parametrize('test_item', TEST_ITEMS,
+                         ids=[ContainersTestItem.get_pretty_id(ti) for ti in TEST_ITEMS])
+def test_smart_management_add_tag(appliance, provider, test_item):
+
     # validate no tag set to project
-    obj_inst = (provider if test_item.obj is ContainersProvider
-                else test_item.collection_obj(appliance).get_random_instances().pop())
-
-    logger.debug('Selected object is "{obj_name}"'.format(obj_name=obj_inst.name))
-
-    regex = r"([\w\s|\-|\*]+:([\w\s|\-|\*])+)|(No.*assigned)"
-    try:
-        # Remove all previous configured tags for given object
-        logger.debug('Starting cleaning old tags from '
-                    'object "{obj_name}"'.format(obj_name=obj_inst.name))
-        obj_inst.remove_tags(obj_inst.get_tags())
-        logger.debug("All smart management tags was removed successfully")
-    except RuntimeError:
-        logger.debug("Fail to remove tags, checking if no tag set")
-
-        # Validate old tags formatting
-        assert re.match(regex, wait_for_tag(obj_inst).text_value), (
-            "Tag formatting is invalid! ")
-        logger.debug("No tag was set, continuing to main test")
-
-    # Config random tag for object\
-    random_tag_set = set_random_tag(obj_inst)
-
-    logger.debug("Fetching tag info for selected object")
-    # validate new tag format
-    tag_display_text = wait_for_tag(obj_inst)
-    logger.debug("Tag info: {info}".format(info=tag_display_text))
-
-    assert re.match(regex, tag_display_text), "Tag formatting is invalid! "
-    actual_tags_on_instance = obj_inst.get_tags()
-
-    # Validate tag seted successfully
-    assert len(actual_tags_on_instance) == 1, "Fail to set a tag for {obj_type}".format(
-        obj_type=get_object_name(test_item.obj))
-    actual_tags_on_instance = actual_tags_on_instance.pop()
+    obj_collection = getattr(appliance.collections, test_item.collection_obj)
+    obj_inst = get_clean_entity(entity=provider) \
```

Comment:
Required: use parenthesis here
obj_inst = (get_clean_entity(entity=provider) 
if test_item.obj is ContainersProvider else get_clean_entity(collection=obj_collection))

Reply:
@rlbabyuk This code is not part of the PR anymore
you can see the latest version of this code here --> [link]( https://github.com/shalomnaim1/integration_tests/blob/0af82c8b9fdad8a0a05efc68ed230b5bbf89a4a3/cfme/tests/containers/test_smart_management.py)

Reply:
@shalomnaim1 the same change I also can see https://github.com/shalomnaim1/integration_tests/blob/0af82c8b9fdad8a0a05efc68ed230b5bbf89a4a3/cfme/tests/containers/test_smart_management.py#L78